### ISSUE TITLE
Popup Options

### ISSUE THREAD

**Issue by CraigChamberlain** [beyonk-adventures/svelte-mapbox#40]
Atm, the following options are listed by name.  

```
 const popupEl = new mapbox.Popup({
  offset: popupOffset,
  className: popupClassName
})
```
There are a good number of other options in the API in the [Docs](https://docs.mapbox.com/mapbox-gl-js/api/markers/#popup)

My problem here is that naming all of them in the marker is going to get a bit excessive.  If I were to merge popupOffset & popupClassName into and popupOptions object this would be a breaking change. :( 

What are the implications of spreading an object
```
const popupEl = new mapbox.Popup({
  offset: popupOffset,
  className: popupClassName,
  ...popupOptions
})
```

Would something like this be better

```
namedParams = { offset: popupOffset,  className: popupClassName }
const popupEl = new mapbox.Popup(Object.assign(namedParams, popupOptions))
```

**Comment by antony** [comment]
I think the spread option is fine, but to retain compatibility with crap browsers, it's better to use Object.assign

**Comment by CraigChamberlain** [comment]
Just working on it now.  Will pop a PR together.

**Comment by antony** [comment]
Released as v7.7.1
